In [3]:
# mount Google Drive
import pandas as pd
from google.colab import drive
import seaborn as sns
import matplotlib as plt

In [4]:
# import the drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


# Exploratory Data Analysis of the 'articles.csv' dataset
We search for missing values, for values we did not expect such as outliers or special values.

In [5]:
# import the csv file, read it and put it into a DataFrame
gdrive_path_articles = '/content/gdrive/My Drive/Univ/AI_Project/articles.csv'
articles = pd.read_csv(gdrive_path_articles)
articles = pd.DataFrame(articles)

In [6]:
articles

article_id  product_code               prod_name  product_type_no  \
0        108775015        108775               Strap top              253   
1        108775044        108775               Strap top              253   
2        108775051        108775           Strap top (1)              253   
3        110065001        110065       OP T-shirt (Idro)              306   
4        110065002        110065       OP T-shirt (Idro)              306   
...            ...           ...                     ...              ...   
105537   953450001        953450  5pk regular Placement1              302   
105538   953763001        953763       SPORT Malaga tank              253   
105539   956217002        956217         Cartwheel dress              265   
105540   957375001        957375        CLAIRE HAIR CLAW               72   
105541   959461001        959461            Lounge dress              265   

       product_type_name  product_group_name  graphical_appearance_no  \
0               Vest top  Garment Upper body                  1010016   
1               Vest top  Garment Upper body                  1010016   
2               Vest top  Garment Upper body                  1010017   
3                    Bra           Underwear                  1010016   
4                    Bra           Underwear                  1010016   
...                  ...                 ...                      ...   
105537             Socks      Socks & Tights                  1010014   
105538          Vest top  Garment Upper body                  1010016   
105539             Dress   Garment Full body                  1010016   
105540         Hair clip         Accessories                  1010016   
105541             Dress   Garment Full body                  1010016   

       graphical_appearance_name  colour_group_code colour_group_name  ...  \
0                          Solid                  9             Black  ...   
1                          Solid                 10             White  ...   
2                         Stripe                 11         Off White  ...   
3                          Solid                  9             Black  ...   
4                          Solid                 10             White  ...   
...                          ...                ...               ...  ...   
105537           Placement print                  9             Black  ...   
105538                     Solid                  9             Black  ...   
105539                     Solid                  9             Black  ...   
105540                     Solid                  9             Black  ...   
105541                     Solid                 11         Off White  ...   

          department_name index_code        index_name index_group_no  \
0            Jersey Basic          A        Ladieswear              1   
1            Jersey Basic          A        Ladieswear              1   
2            Jersey Basic          A        Ladieswear              1   
3          Clean Lingerie          B  Lingeries/Tights              1   
4          Clean Lingerie          B  Lingeries/Tights              1   
...                   ...        ...               ...            ...   
105537          Socks Bin          F          Menswear              3   
105538             Jersey          A        Ladieswear              1   
105539             Jersey          A        Ladieswear              1   
105540  Small Accessories          D           Divided              2   
105541             Jersey          A        Ladieswear              1   

        index_group_name section_no            section_name garment_group_no  \
0             Ladieswear         16  Womens Everyday Basics             1002   
1             Ladieswear         16  Womens Everyday Basics             1002   
2             Ladieswear         16  Womens Everyday Basics             1002   
3             Ladieswear         61         Womens Lingerie             

In [7]:
# let's check if any values are missing. As we do not yet want to search a specific column we just look at the entire dataframe
articles.isna().sum()


article_id                        0
product_code                      0
prod_name                         0
product_type_no                   0
product_type_name                 0
product_group_name                0
graphical_appearance_no           0
graphical_appearance_name         0
colour_group_code                 0
colour_group_name                 0
perceived_colour_value_id         0
perceived_colour_value_name       0
perceived_colour_master_id        0
perceived_colour_master_name      0
department_no                     0
department_name                   0
index_code                        0
index_name                        0
index_group_no                    0
index_group_name                  0
section_no                        0
section_name                      0
garment_group_no                  0
garment_group_name                0
detail_desc                     416
dtype: int64

Clearly the values are well kept and only the detail description lacks a bit. In total 416 values are missing.
Now let's look at the descriptions of the columns. We want to look for anomalies, and values out of the ordinary.

In [10]:
# using this we get lots of info about counts, means, deviations, and so forth ...

articles["prod_name"].describe()

count              105542
unique              45875
top       Dragonfly dress
freq                   98
Name: prod_name, dtype: object

In [15]:
# lets look at the description of both the product code and the product type number
articles["product_code"].value_counts()

783707    75
684021    70
699923    52
699755    49
685604    46
          ..
761368     1
761366     1
761360     1
761350     1
959461     1
Name: product_code, Length: 47224, dtype: int64

In [16]:
articles["product_type_no"].value_counts()

272    11169
265    10362
252     9302
255     7904
254     4155
       ...  
525        1
514        1
351        1
349        1
483        1
Name: product_type_no, Length: 132, dtype: int64

In [17]:
articles["product_type_name"].value_counts()

Trousers         11169
Dress            10362
Sweater           9302
T-shirt           7904
Top               4155
                 ...  
Keychain             1
Headband             1
Cushion              1
Blanket              1
Clothing mist        1
Name: product_type_name, Length: 131, dtype: int64

One thing we can notice here is that the length differs with one. We can ask ourselves if we maybe missed a value, but as we have seen earlier, no value is missing.

In [18]:
pr_ty_no = articles["product_type_no"].isna().sum()
pr_ty_na = articles["product_type_name"].isna().sum()

pr_ty_no, pr_ty_na

(0, 0)